File: app/src/main/java/dev/pitlor/smssync/viewmodels/MessageFragmentViewModel.java
package dev.pitlor.smssync.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.hilt.lifecycle.ViewModelInject;
import androidx.lifecycle.LiveData;

public class MessageFragmentViewModel extends BaseViewModel {
    public LiveData<Integer> messagesCount = appRepository.getMessageCount();

    @ViewModelInject
    public MessageFragmentViewModel(@NonNull Application application) {
        super(application);
    }

    public boolean isLoading() {
        return messagesCount == null;
    }
}

File: app/src/main/java/dev/pitlor/smssync/viewmodels/MessageFragmentViewModel.kt
package dev.pitlor.smssync.viewmodels

import android.app.Application
import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.LiveData
import dev.pitlor.smssync.datasources.AppRepository

class MessageFragmentViewModel @ViewModelInject constructor(appRepository: AppRepository, application: Application) : BaseViewModel(application) {
    var messagesCount: LiveData<Int> = appRepository.messageCount
}